jgiven-core/src/main/java/com/tngtech/jgiven/report/AbstractReport.java

```diff
@@ -0,0 +1,184 @@
+
+    private List<String> flags = new ArrayList<String>(
+            Arrays.asList( "-h", "--help", "--format=", "--sourceDir=", "--targetDir=", "--title=", "--exclude-empty-scenarios=", "--dir",
+                    "--todir" ) );
+    private Map<String, String> flagMap;
+    private String format;
+    private String title;
+    private File sourceDir;
+    private File targetDir;
+    private Boolean excludeEmptyScenarios;
+
+    public String getFormat() {
+        return format;
+    }
+
+    private void setFormat( String format ) {
+        this.format = format;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    private void setTitle( String title ) {
+        this.title = title;
+    }
+
+    public File getSourceDir() {
+        return sourceDir;
+    }
+
+    public void setSourceDir( String pathname ) {
+        this.sourceDir = new File( pathname );
+    }
+
+    public File getTargetDir() {
+        return targetDir;
+    }
+
+    public void setTargetDir( String pathname ) {
+        this.targetDir = new File( pathname );
+    }
+
+    public Boolean getExcludeEmptyScenarios() {
+        return excludeEmptyScenarios;
+    }
+
+    private void setExcludeEmptyScenarios( Boolean excludeEmptyScenarios ) {
+        this.excludeEmptyScenarios = excludeEmptyScenarios;
+    }
+
+    public void addFlags( String... flags ) {
+        for( String flag : flags ) {
+            this.flags.add( flag );
+        }
+    }
+
+    public void parseFlags() {
+        if( flagMap.containsKey( "-h" ) || flagMap.containsKey( "--help" ) ) {
+            printUsageAndExit();
+        }
+        if( !flagMap.containsKey( "--format=" ) ) {
+            System.err.println( "No argument provided for --format. Terminating." );
+            printUsageAndExit();
+        } else {
+            setFormat( flagMap.get( "--format=" ) );
+        }
+        setTitle( getOrDefault( flagMap, "--title=", "JGiven Report" ) );
+        setSourceDir( getOrDefault( flagMap, "--sourceDir=", "." ) );
+        setTargetDir( getOrDefault( flagMap, "--targetDir=", "." ) );
+
+        if( flagMap.containsKey( "--dir=" ) ) {
+            setSourceDir( flagMap.get( "--dir=" ) );
+            System.err.println( "DEPRECATION WARNING: --dir is deprecated, please use --sourceDir instead" );
+        }
+        if( flagMap.containsKey( "--todir=" ) ) {
+            setTargetDir( flagMap.get( "--todir=" ) );
+            System.err.println( "DEPRECATION WARNING: --todir is deprecated, please use --targetDir instead" );
+        }
+
+        setExcludeEmptyScenarios( Boolean.parseBoolean( getOrDefault( flagMap, "--exclude-empty-scenarios=", "false" ) ) );
+        parseFlags( flagMap );
+    }
+
+    public void printUsageAndExit() {
+        System.err
+                .println( "Options: \n"
+                        + "  --format=<format>             the format of the report. Either html or text (required) \n"
```

Comment:
It could be overkill, but I wonder whether you could not generalize your code further by introducing an CommandLineOption class that contains all information about a command line option, like description, deprecation and so on. Then you code will become more general in the end.